tests/Integration/PrestaShopBundle/Translation/Extractor/LegacyModuleExtractorTest.php

```diff
@@ -30,65 +30,106 @@
 {
     /**
      * @var string Domain name of the modules translations
      */
-    const DOMAIN_NAME = 'ModulesSomeModule';
+    const DOMAIN_NAME = 'ModulesTranslationtest';
+    const MODULE_NAME = 'translation_test';
 
     /**
-     * @cover extract
+     * @var CatalogueVerifier
      */
-    public function testLegacyModuleExtractorShouldReturnACatalogue()
-    {
-        self::bootKernel();
-        $phpExtractor = new PhpExtractor();
-        $smartyExtractor = self::$kernel->getContainer()->get('prestashop.translation.extractor.smarty');
-        $extractor = new LegacyModuleExtractor($phpExtractor, $smartyExtractor, $this->getModuleFolder());
+    private $catalogueVerifier;
 
-        $catalogue = $extractor->extract('some_module', 'fr-FR');
+    public function __construct($name = null, array $data = [], $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
 
-        $this->assertInstanceOf(MessageCatalogueInterface::class, $catalogue);
+        $this->catalogueVerifier = new CatalogueVerifier($this);
     }
 
     /**
-     * @depends testLegacyModuleExtractorShouldReturnACatalogue
+     * @param string $locale
+     * @param array $expected
+     *
+     * @cover extract
+     * @dataProvider provideTestCases
      */
-    public function testExtractedCatalogueContainsTheExpectedTranslations()
+    public function testExtractedCatalogueContainsTheExpectedWordings($locale, $expected)
     {
         self::bootKernel();
-        $phpExtractor = new PhpExtractor();
-        $smartyExtractor = self::$kernel->getContainer()->get('prestashop.translation.extractor.smarty');
-        $extractor = new LegacyModuleExtractor($phpExtractor, $smartyExtractor, $this->getModuleFolder());
+        $container = self::$kernel->getContainer();
 
-        $catalogue = $extractor->extract('some_module', 'fr-FR');
-        $this->assertCount(5, $catalogue->all(self::DOMAIN_NAME));
-        $this->assertTrue($catalogue->has(
-            'An error occured, please check your zip file',
-            self::DOMAIN_NAME
-        ));
+        $phpExtractor = $container->get('prestashop.translation.extractor.php');
+        $smartyExtractor = $container->get('prestashop.translation.extractor.smarty.legacy');
+        $twigExtractor = $container->get('prestashop.translation.extractor.twig');
 
-        $this->assertSame(
-            'The module %s has been disabled',
-            $catalogue->get('The module %s has been disabled', self::DOMAIN_NAME)
+        $extractor = new LegacyModuleExtractor(
+            $phpExtractor,
+            $smartyExtractor,
+            $twigExtractor,
+            $this->getModuleFolder()
         );
-    }
 
-    protected function tearDown()
-    {
-        self::$kernel->shutdown();
+        $catalogue = $extractor->extract(self::MODULE_NAME, $locale);
+
+        $this->assertInstanceOf(MessageCatalogueInterface::class, $catalogue);
+
+        $this->catalogueVerifier->verify($catalogue, $expected);
```

Comment:
this makes the tests less readable as we now have to look at CatalogueVerifier to figure out what has failed here, this remembers me how many hours I've already lost figure out how works the old Translation tests.

Please, don't do that: if you want to test behaviors, rely on Behat instead.

Reply:
The catalogue verifier is very verbose when it fails, but I can change the method name to make it more evident about what it does. It just verifies that the catalogue tree matches the provided one, that's all.